internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -559,6 +559,27 @@ func StartContainerd(seq runtime.Sequence, data interface{}) (runtime.TaskExecut
 	}, "startContainerd"
 }
 
+// WriteUdevRules is the task that writes udev rules to a udev rules file.
+func WriteUdevRules(seq runtime.Sequence, data interface{}) (runtime.TaskExecutionFunc, string) {
+	return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) (err error) {
+		var result *multierror.Error
+
+		rules := r.Config().Machine().Udev().Rules()
+
+		content := ""
+
+		for _, rule := range rules {
+			content += strings.ReplaceAll(rule, "\n", "\\\n") + "\n"
+		}
+
+		if err = ioutil.WriteFile(constants.UdevRulesPath, []byte(content), 0o644); err != nil {
+			result = multierror.Append(result, err)
+		}
+
+		return result.ErrorOrNil()
+	}, "writeUdevRules"
+}
```

Comment:
Do you happen to have any pointers for a noob to write some tests for this? I was expecting to find a `v1alpha1_sequencer_tasks_test.go`, but I didn't see one. So not sure if we just have integration tests somewhere or something that cover most of these functions.

Reply:
this will be a bit hard to test actually the way it is implemented right now

we certainly feel we need this feature, but we're not sure that it's the best way to implement that. I'll take look a bit more deeply in the coming days and get back to you with some ideas.

Reply:
Sounds good! I open to other options!